### PR TITLE
Feat plugin mutable comment

### PR DESCRIPTION
**Pull request by kwonoj**
**Description:**

PR implements remaining comments interfaces along with some refactor to reduce repeated codes for the implementations. Also separate hostenvironment for each impl to not to expose too much details for other features.

**BREAKING CHANGE:**

**Related issue (if exists):**

- closes https://github.com/swc-project/swc/issues/4182

### CODE REVIEW

File: crates/swc_plugin_comments/src/plugin_comments_proxy.rs [hunk 1]
```diff
@@ -81,54 +111,135 @@ impl Comments for PluginCommentsProxy {
...
+}
+
+#[cfg(feature = "plugin-mode")]
+#[cfg_attr(not(target_arch = "wasm32"), allow(unused))]
+impl Comments for PluginCommentsProxy {
+    fn add_leading(&self, pos: BytePos, cmt: Comment) {
+        self.allocate_comments_buffer_to_host(&cmt);
+        #[cfg(target_arch = "wasm32")]
+        unsafe {
+            __add_leading_comment_proxy(pos.0);
```
Comment (kdy1) [inline crates/swc_plugin_comments/src/plugin_comments_proxy.rs:129]: How does this work? We don't pass the pointer of the host memory we allocated above?

Comment (kwonoj) [reply crates/swc_plugin_comments/src/plugin_comments_proxy.rs:129]: we do, https://github.com/swc-project/swc/pull/4229/files#diff-5d0fdaffb501760b81888c7b39468d14561cb4f509dd056f2f68ebdfaf3ead93R45

right above does the job. this is same way to set tranaform result, using hostenv's attached buffer guest copies value into host.

Comment (kdy1) [reply crates/swc_plugin_comments/src/plugin_comments_proxy.rs:129]: Yeah, but I mean I'm not sure how `__add_leading_comment_proxy` works without `&pointer_to_comment_in_host_memory`

Comment (kdy1) [reply crates/swc_plugin_comments/src/plugin_comments_proxy.rs:129]: Ah, I found `mutable_comment_buffer`

**Review by kdy1 — APPROVED** [review]
Thank you!

---

swc-bump:
 - swc_common
 - swc_plugin --breaking
 - swc_plugin_runner --breaking

**Pull request merged**